### ISSUE TITLE
AuthenticationManager missing when using spring.security grpc-spring-boot-starter

### ISSUE THREAD

**Issue by Wumba** [LogNet/grpc-spring-boot-starter#361]
In my Project, multiple components use gRPC to communicate with each other. 
I am using OpenApi to generate java client code. 
This offers an MergedApi.
```java
@Bean
public MergedApi mergedApi(final ApiClient apiClient) {
  return new MergedApi(apiClient);
}
```

To add my token to the ApiClient, I need to use the following dependencies: 

```xml
   <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-security</artifactId>
            <version>${spring-boot-starter-security.version}</version>
        </dependency>
```
Unfortunately needed because spring boot does not offer all the necessary classes to shove my token down MergedApi's throat 
```xml
        <dependency>
            <groupId>org.springframework.security</groupId>
            <artifactId>spring-security-oauth2-client</artifactId>
            <version>${spring-security-oauth2-client.version}</version>
        </dependency>
```
Needed for enabling spring boot to make calls to REST-Endpoints
```xml
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-oauth2-client</artifactId>
            <version></version>
        </dependency>
```

I guess you all know this dependency:
```xml
            <dependency>
                <groupId>io.github.lognet</groupId>
                <artifactId>grpc-spring-boot-starter</artifactId>
                <version>${spring-boot-grpc-starter.version}</version>
                <scope>compile</scope>
            </dependency>
```

When I start the application I get the message: 
``` 
Error creating bean with name 'springGrpcSecurityInterceptor' defined in class path resource [org/lognet/springboot/grpc/security/GrpcSecurityConfiguration.class]: Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: An AuthenticationManager is required
```

On my debugging journey to hell and back, I stumbled across this beauty in `org.lognet.springboot.grpc.security.GrpcSecurityConfigurerAdapter`: 
```java 
  @Override
  public void configure(GrpcSecurity builder) throws Exception {
    try {
      final Class<?> jwtDecoderClass =
          Class.forName("org.springframework.security.oauth2.jwt.JwtDecoder");
      final String[] beanNames = context.getBeanNamesForType(jwtDecoderClass);
      if (1 == beanNames.length) {
        builder.authenticationProvider(
            JwtAuthProviderFactory.forAuthorities(context.getBean(beanNames[0], JwtDecoder.class)));
      }
    } catch (ClassNotFoundException e) {
    }
    // swallow
    builder.authorizeRequests().withSecuredAnnotation();
  }
```
So apparently when no JwtDecoder-Bean exists, the whole Application crashes and this error is thrown somewhere else since this catch block is empty.
I added manually the JwtDecoder and the necessary dependency for it: 

```java
    @Bean
    public JwtDecoder decoder() {
        return token -> Jwt.withTokenValue(token).build();
    }
```
```xml
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
            <version>${spring-boot-starter-security.version}</version>
        </dependency>
```
Why does org.lognet.springboot.grpc.security interact this way with spring-security-oauth2-client? Am I missing something? 
Why do I need a JwtDecoder here (apart from being mandatory in the code)? 
I am very confused.

**Comment by jvmlet** [comment]
Do you want your grpc services to be secured ?

**Comment by Wumba** [comment]
> Do you want your grpc services to be secured ?

Actually no, the gRPC communication in the project is not secured or encrypted as far as I know. It's running on a separated server, so we have other problems if somebody is able to listen to the communication. 

The second application I am trying to access is not mine, therefore I cannot disable the use of REST and switch to gRPC. 

The openapi-tools-generator generates a spring webclient using spring.webflux. Might that be an issue? 
I've read that grpc-spring-boot-starter uses spring.web itself under the hood? Is that true? Showing the dependency tree with maven did not show me any signs of this.

**Comment by Wumba** [comment]
Thank you very much, I figured that out one hour before. I guess I was just to dumb to read and understand the sentence: 
> Autoconfigures and runs the embedded gRPC server with @GRpcService-enabled beans as part of spring-boot application
> By setting grpc.security.auth.enabled to false, GRPC security can be turned-off.

I did not connect the dots between AuthenticationManager and the use of spring-security-config. 
I will never forget, debugged to hell and back. Maybe I gonna write a pull request to add a helpful message in the catch block in the `GrpcSecurityConfigurerAdapter#configure` like "maybe you are using spring-security-config without disabling grpc-security auth? "
